File: src/inventory-service/inventory-core/src/main/java/com/inventory/core/EventPublisher.java
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2024 Datadog, Inc.
 */

package com.inventory.core;

public interface EventPublisher {
    void publishNewProductAddedEvent(NewProductAddedEvent evt);

    void publishInventoryStockUpdatedEvent(InventoryStockUpdatedEvent evt);

    void publishStockReservedEvent(StockReservedEventV1 evt);

    void publishProductOutOfStockEvent(ProductOutOfStockEventV1 evt);

    void publishStockReservationFailedEvent(StockReservationFailedEventV1 evt);
}

File: src/order-service/src/Orders.Core/EventPublisher.cs
// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2025 Datadog, Inc.

using Orders.Core.PublicEvents;

namespace Orders.Core;

public interface IEventGateway
{
    Task HandleOrderCreated(Order order);
    Task HandleOrderConfirmed(Order order);
    Task HandleOrderCompleted(Order order);
}